aiocoap/protocol.py

```diff
@@ -159,7 +159,7 @@ async def create_client_context(cls, *, loggername="coap", loop=None, transports
             if transportname == 'udp6':
                 from .transports.udp6 import MessageInterfaceUDP6
                 await self._append_tokenmanaged_messagemanaged_transport(
-                    lambda mman: MessageInterfaceUDP6.create_client_transport_endpoint(mman, log=self.log, loop=loop))
+                    lambda mman: MessageInterfaceUDP6.create_client_transport_endpoint(mman, log=self.log, loop=loop, broadcast=broadcast))
```

Comment:
Is there any good reason to make this a flag? IOW: Will anything bad come of aiocoap always setting that option?

We're setting a lot of socket options  even though we don't know yet whether we'll need them on this particular occasion. (For example, V6ONLY=0 is set even though we may not use IPv4 at all).

Reply:
Very good question. I *think* the downside is that superuser privileges (or possibly some CAP_* on Linux) are required to set this particular ioctl, but I need to check on that

Reply:
I think I'm misremembering regarding the requirement of additional privileges, the EPERM was coming from an earlier version of this patch when I was using SO_BINDTODEVICE

You may wonder why on earth that would be necessary- the server software I was dealing with actually _required_ the universal broadcast address 255.255.255.255 for a specific endpoint, and I'm working in a multi-homed environment with multiple interfaces having +BROADCAST

In the end I decided it would be outside the scope of this PR to add an option to force the interface, especially without any discussion. My final conclusion was to not attempt to solve that problem with changes to aiocoap; there are a few workarounds good enough for that exceptional case

tl; dr; your suggestion on defaulting to SO_BROADCAST seems to be a good one as far as I can tell